NSFS on syscall error should log the operation info and not just the error code (even if debug level is low)

### Environment info
- NooBaa Version: **5.15.3**
- Platform: **rhel**

### Actual behavior
1. on syscall error we only see the err code unless we turn on higher debug log -
```
May 21 16:19:48 host node[2941584]: May-21 16:19:48.654 [nsfs/2941584] [ERROR] core.sdk.namespace_fs::  [Error: Operation not permitted] { code: 'EPERM' }
```

### Expected behavior
1. Should log the operation info with the error code.

### Steps to reproduce
1. Change fs permissions to cause EPERM.

--- a/src/endpoint/s3/s3_rest.js
+++ b/src/endpoint/s3/s3_rest.js
@@ -399,7 +399,9 @@ function handle_error(req, res, err) {
     dbg.error('S3 ERROR', reply,
         req.method, req.originalUrl,
         JSON.stringify(req.headers),
-        err.stack || err);
+        err.stack || err,
+        err.context ? `- context: ${err.context?.trim()}` : '',
+    );
     if (res.headersSent) {
         dbg.log0('Sending error xml in body, but too late for headers...');
     } else {

--- a/src/native/fs/fs_napi.cpp
+++ b/src/native/fs/fs_napi.cpp
@@ -655,6 +655,9 @@ struct FSWorker : public Napi::AsyncWorker
         if (_errno) {
             obj.Set("code", Napi::String::New(env, uv_err_name(uv_translate_sys_error(_errno))));
         }
+
+        obj.Set("context", Napi::String::New(env, _desc));
+
         _deferred.Reject(obj);
     }
 };
@@ -1311,7 +1314,7 @@ struct Fsync : public FSWorker
 };
 
 /**
- * GetPwName is an os op 
+ * GetPwName is an os op
  */
 struct GetPwName : public FSWorker
 {
@@ -2206,7 +2209,7 @@ fs_napi(Napi::Env env, Napi::Object exports)
             if (sizeof(struct gpfsRequest_t) != 256) {
                 PANIC("The gpfs get extended attributes is of wrong size" << sizeof(struct gpfsRequest_t));
             }
-            
+
             auto gpfs = Napi::Object::New(env);
             gpfs["register_gpfs_noobaa"] = Napi::Function::New(env, register_gpfs_noobaa);
             // we export the gpfs object, which can be checked to indicate that

--- a/src/native/util/common.h
+++ b/src/native/util/common.h
@@ -23,7 +23,11 @@ namespace noobaa
 {
 
 #ifndef __func__
-#define __func__ __FUNCTION__
+    #ifdef __PRETTY_FUNCTION__
+    #define __func__ __PRETTY_FUNCTION__
+    #else
+    #define __func__ __FUNCTION__
+    #endif
 #endif
 
 #define DVAL(x) #x "=" << x << " "

